### PR TITLE
Add Edge Identity API Reference

### PR DESCRIPTION
**Pull request by kevinlind**
Add Edge Identity API Reference for all public APIs and classes. Adds resetIdentities API to Mobile Core API reference page.

### CODE REVIEW

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+```
+{% endtab %}
+{% endtabs %}
+
+## updateIdentities
+{% tabs %}
+{% tab title="Android" %}
+### updateIdentities
+
+Update the currently known identites within the SDK. The Edge Identity extension will merge the received identifiers with the previously saved ones in an additive manner, no identities are removed from this API.
```
Comment (nporter-adbe) [inline using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:151]: identites -> identities

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+
+Identities with an empty _id_ or _namespace_ are not allowed and are ignored.
+
+Updating identites using a reserved namespace is not allowed using this API. The reserved namespaces are:
```
Comment (nporter-adbe) [inline using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:155]: identites -> identities

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+IdentityMap identityMap = new IdentityMap();
+identityMap.addItem(item, "Email")
+Identity.updateIdentities(identityMap);
+```
+{% endtab %}
+
+{% tab title="iOS" %}
+### updateIdentities
+
+Update the currently known identites within the SDK. The Edge Identity extension will merge the received identifiers with the previously saved ones in an additive manner, no identities are removed from this API.
```
Comment (nporter-adbe) [inline using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:183]: identites -> identities

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+
+Identities with an empty _id_ or _namespace_ are not allowed and are ignored.
+
+Updating identites using a reserved namespace is not allowed using this API. The reserved namespaces are:
```
Comment (nporter-adbe) [inline using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:187]: identites -> identities

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md [hunk 5]
```diff
@@ -0,0 +1,520 @@
...
+## removeIdentity
+{% tabs %}
+{% tab title="Android" %}
+### removeIdentity
+
+Remove the identity from the stored client-side [IdentityMap](#identitymap). The Identity extension will stop sending the identifier to the Edge Network. Using this API does not remove the identifier from the server-side User Profile Graph or Identity Graph.
+
+Identities with an empty _id_ or _namespace_ are not allowed and are ignored.
+
+Removing identites using a reserved namespace is not allowed using this API. The reserved namespaces are:
```
Comment (nporter-adbe) [inline using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:230]: identites -> identities

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md [hunk 6]
```diff
@@ -0,0 +1,520 @@
...
+{% endtab %}
+
+{% tab title="iOS" %}
+### removeIdentity
+
+Remove the identity from the stored client-side [IdentityMap](#identitymap). The Identity extension will stop sending the identifier to the Edge Network. Using this API does not remove the identifier from the server-side User Profile Graph or Identity Graph.
+
+Identities with an empty _id_ or _namespace_ are not allowed and are ignored.
+
+Removing identites using a reserved namespace is not allowed using this API. The reserved namespaces are:
```
Comment (nporter-adbe) [inline using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:260]: identites -> identities

File: using-mobile-extensions/mobile-core/mobile-core-api-reference.md [hunk 7]
```diff
@@ -635,6 +639,64 @@ MobileCore.getSdkIdentities { (content, error) in
 {% endtab %}
 {% endtabs %}
 
+## resetIdentities
+
+Clears the identites from Edge extensions and resets the Experience Cloud ID (ECID).
```
Comment (nporter-adbe) [inline using-mobile-extensions/mobile-core/mobile-core-api-reference.md:644]: identites -> identities

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md [hunk 8]
```diff
@@ -0,0 +1,520 @@
...
+
+// Remove an item
+IdentityItem item = new IdentityItem("user@example.com");
+identityMap.removeItem(item, "Email");
+
+// Get list of items for a given namespace
+List<IdentityItem> items = identityMap.getIdentityItemsForNamespace("Email");
+
+// Get list of all namespaces used in current IdentityMap
+List<String> namespaces = identityMap.getNamespaces();
```
Comment (nporter-adbe) [inline using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:361]: Get a list?

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md [hunk 9]
```diff
@@ -0,0 +1,520 @@
...
+
+// Add an item
+IdentityItem item = new IdentityItem("user@example.com");
+identityMap.addItem(item, "Email");
+
+// Remove an item
+IdentityItem item = new IdentityItem("user@example.com");
+identityMap.removeItem(item, "Email");
+
+// Get list of items for a given namespace
```
Comment (nporter-adbe) [inline using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:357]: Get a list?

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md [hunk 10]
```diff
@@ -0,0 +1,520 @@
...
+
+**Java**
+
+**Syntax**
+
+```java
+public static void getExperienceCloudId(final AdobeCallback<String> callback);
+```
+
+* _callback_ is invoked after the ECID is available.
```
Comment (nporter-adbe) [inline using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:23]: Should we specify callbacks can be called on different threads? I think we do in our other docs.

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md [hunk 11]
```diff
@@ -0,0 +1,520 @@
...
+This ID is preserved between app upgrades, is saved and restored during the standard application backup process, and is removed at uninstall. The values are returned via the callback.
+
+**Syntax**
+
+```swift
+@objc(getExperienceCloudId:)
+static func getExperienceCloudId(completion: @escaping (String?, Error?) -> Void)
+```
+
+* _completion_ is invoked after the ECID is available.  The default timeout is 1000ms.
```
Comment (nporter-adbe) [inline using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:51]: I'm on the fence about whether or not we need to specify the timeout in the API callbacks...

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md [hunk 12]
```diff
@@ -0,0 +1,520 @@
...
+}];
+```
+
+**Swift**
+
+```swift
+Identity.getExperienceCloudId { (ecid, error) in
+  if let error = error {
+    // handle error here
+  } else {
```
Comment (nporter-adbe) [inline using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:69]: `ecid` can still be nil here. Should we also unwrap `ecid`?

`} else if let ecid = ecid { ... }`

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md [hunk 13]
```diff
@@ -0,0 +1,520 @@
...
+}];
+```
+
+**Swift**
+
+```swift
+Identity.getIdentities { (identityMap, error) in
+  if let error = error {
+    // handle error here
+  } else {
```
Comment (nporter-adbe) [inline using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:138]: Same question here about unwrapping the identity map as it's marked as optional.

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md [hunk 14]
```diff
@@ -0,0 +1,520 @@
...
+identityMap.add(item: IdentityItem(id: "user@example.com"), withNamespace: "Email")
+
+// Remove an item
+identityMap.remove(item: IdentityItem(id: "user@example.com", withNamespace: "Email"))
+
+// Get list of items for a given namespace
+let items: [IdentityItem] = identityMap.getItems(withNamespace: "Email")
+
+// Get list of all namespaces used in current IdentityMap
+let namespaces: [String] = identityMap.namespaces
```
Comment (nporter-adbe) [inline using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:411]: Get a list?

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md [hunk 15]
```diff
@@ -0,0 +1,520 @@
...
+let identityMap: IdentityMap = IdentityMap()
+
+// Add an item
+identityMap.add(item: IdentityItem(id: "user@example.com"), withNamespace: "Email")
+
+// Remove an item
+identityMap.remove(item: IdentityItem(id: "user@example.com", withNamespace: "Email"))
+
+// Get list of items for a given namespace
+let items: [IdentityItem] = identityMap.getItems(withNamespace: "Email")
```
Comment (nporter-adbe) [inline using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:408]: Get a list?

**Review by nporter-adbe — CHANGES_REQUESTED** [review]

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md (hunk 11 above)
Comment (kevinlind) [reply using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:51]: I took this from the Identity API docs. We have had several customer questions about the timeout used for `getExperienceCloudId`, so I think we should explicitly state it.

File: using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md (hunk 12 above)
Comment (kevinlind) [reply using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md:69]: This example is to show the user should check if an error exists before using the returned `ecid` value, with just a generic statement of "handle the retrieved identifiers here" if there is no error. I don't want to bog down the example with too many checks. The syntax example does state they value is optional. Conversely, I could just go with,
```
Identity.getExperienceCloudId { (ecid, error) in
   // handle ecid 
}
```

**Review by nporter-adbe — APPROVED** [review]

**Pull request merged**